scripts/actions/client.js

```diff
@@ -0,0 +1,207 @@
+import KintoClient from "kinto-client";
+import { updatePath } from "redux-simple-router";
+
+import { notifyError, notifySuccess } from "./notifications";
+import * as SessionActions from "./session";
+import * as CollectionActions from "./collection";
+import * as RecordActions from "./record";
+import { CLIENT_BUSY } from "../constants";
+
+
+export let client;
+
+function getClient({session: {server, username, password}}) {
+  return client || new KintoClient(server, {
+    headers: {
+      Authorization: "Basic " + btoa([username, password].join(":")),
+      // XXX by default we must avoid caching authentication-dependent resources
+      // as they could leak private information to different users. Here we
+      // simply discard caching entirely, waiting for a Kinto server fix.
+      // ref https://github.com/Kinto/kinto/issues/593
+      Pragma: "no-cache",
+    }
+  });
+}
+
+export function resetClient() {
+  client = null;
+}
+
+export function clientBusy(busy) {
+  return {
+    type: CLIENT_BUSY,
+    busy,
+  };
+}
+
+function execute(fn) {
+  return (dispatch, getState) => {
+    dispatch(clientBusy(true));
+    const client = getClient(getState());
+    return fn(client, dispatch, getState)
+      .catch((err) => {
+        dispatch(notifyError(err));
+      })
+      .then(() => {
+        dispatch(clientBusy(false));
+      });
+  };
+}
+
+export function listBuckets() {
+  return execute((client, dispatch, getState) => {
+    return client.fetchServerInfo()
+      .then((serverInfo) => {
+        dispatch(SessionActions.serverInfoLoaded(serverInfo));
+        // XXX We need to first issue a request to the default bucket in order
+        // to create it so we can retrieve the list of buckets.
+        // ref https://github.com/Kinto/kinto/issues/454
+        return client.bucket(serverInfo.user.bucket).getAttributes();
```

Comment:
Be careful, if the `default_bucket` plugin is not installed, there won't be any `user.bucket`. You could inspect the capabilities or add a simple safety check